### PR TITLE
refactor(maas_api): with your shield or on it

### PR DESCRIPTION
**Pull request by ksheedlo**
Addresses comments from #517.

### CODE REVIEW

**Comment by ksheedlo** [comment]
Are Jenkins and Codecov working?

**Comment by glyph** [comment]
Apparently not. No Jenkins on Mimic, though, just Travis...

**Comment by glyph** [comment]
Travis can re-build master just fine, but I can't find any button to re-build the branch. All https://docs.travis-ci.com/user/pull-requests/#My-Pull-Request-isn’t-being-built says is "it must not be mergeable".

**Comment by glyph** [comment]
Making a new PR with the same commit (on a different branch) seems to have fixed it.

File: mimic/rest/maas_api.py [hunk 1]
```diff
@@ -810,13 +867,12 @@ def get_alarm(self, request, tenant_id, entity_id, alarm_id):
         entities = self._entity_cache_for_tenant(tenant_id).entities
 
         try:
-            _assert_alarm_exists(entities, entity_id, alarm_id)
+            alarm = _get_alarm(entities, entity_id, alarm_id)
+            return json.dumps(alarm.to_json())
         except (ObjectDoesNotExist, ParentDoesNotExist) as e:
```
Comment (glyph) [inline mimic/rest/maas_api.py]: This pattern seems to be repeated a few times, and it would be good to refactor it at some point as well.

**Comment by glyph** [comment]
Codecov shows patch coverage is actually good, despite the status check being broken.

**Comment by glyph** [comment]
:+1: looks like a pretty straightforward implementation of the suggested refactoring. Will merge.

**Pull request merged**